src/fsharp/pars.fsy

```diff
@@ -962,13 +962,15 @@ classMemberSpfn:
        SynMemberSig.Inherit ($4,unionRanges (rhs parseState 3) ($4).Range) }
 
   | opt_attributes opt_declVisibility VAL fieldDecl 
-     { if Option.isSome $2 then errorR(Error(FSComp.SR.parsVisibilityDeclarationsShouldComePriorToIdentifier(),rhs parseState 2))
-       let fld = $4 $1 false 
-       SynMemberSig.ValField(fld,rhs2 parseState 3 4) }
+     { let wholeRange = rhs2 parseState 1 4
+       if Option.isSome $2 then errorR (Error (FSComp.SR.parsVisibilityDeclarationsShouldComePriorToIdentifier (), rhs parseState 2))
+       let fld = $4 $1 false wholeRange 
+       SynMemberSig.ValField (fld, wholeRange) }
 
   | opt_attributes opt_declVisibility STATIC VAL fieldDecl 
-     { if Option.isSome $2 then errorR(Error(FSComp.SR.parsVisibilityDeclarationsShouldComePriorToIdentifier(),rhs parseState 2))
-       SynMemberSig.ValField($5 $1 true,rhs2 parseState 3 5) }
+     { let wholeRange = rhs2 parseState 1 5
```

Comment:
We are also including the `opt_attributes` range as well. When there is an error on the val field, will the error range extend itself to include the attributes? If so, I don't know if that would be desirable because attributes can be on the previous line; would mean the error squiggle would show on whitespace.

Reply:
Seems that it will:
```fsharp
let TcAndPublishMemberSpec cenv env containerInfo declKind tpenv memb = 
    match memb with 
    | SynMemberSig.ValField(_, m) -> error(Error(FSComp.SR.tcFieldValIllegalHere(), m))
....
```

Reply:
Yes, including attributes to the declaration range is one of the motivating reasons for these changes.

Reply:
@TIHan this is how it looks in the previous PR for members:


![image](https://user-images.githubusercontent.com/5175830/67927671-6de5cc00-fb76-11e9-93d3-8c1d7e9a26c1.png)


compared to the current highlighting:
![image](https://user-images.githubusercontent.com/5175830/67926883-72a98080-fb74-11e9-9269-f3e085fe1855.png)

The error of course is that member val definitions aren't allowed in classes without constructors, so underlining the whole member definition is probably quite appropriate.

We could improve the rendering of the squiggle to not display on leading whitespace.